crates/subspace-service/src/network.rs

```diff
@@ -178,6 +183,27 @@ where
         protocol_config
     };
 
+    if let Some(offchain_storage) = offchain_storage {
```

Comment:
Why does this need to be done during in a forked version of network creation? We already have a few custom protocols and never needed to modify Substrate's network for this.

Reply:
We have custom protocols indeed (relay request-response), however, we modified the upstream repository fork for it. I think the current approach is less invasive. Please, let me know if you believe we should modify Substrate code instead.

Reply:
How did we modify our fork? I don't think we have done any changes to request/response part of the codebase and as discussed during our call I believe there should be a way to inject request/response protocol without doing any modification in the first place.